File: exercises/101-109_small_problems/easy_01/easy_10.rb
# easy_10.rb - What's my bonus?

def calculate_bonus(salary, bonus)
  return salary / 2 if bonus == true
  0
end

puts calculate_bonus(2800, true) == 1400
puts calculate_bonus(1000, false) == 0
puts calculate_bonus(50000, true) == 25000

=begin
utp:
- 2 args
  - +int and a boolean (t/f)
- if t, bonus is 1/2 salary; if f, bonus is 0
- questions
  - assume salary is even? if not, do I return a float if the salary is odd and
    there is a bonus?

data:
i: int and boolean
o: int

test cases:
puts calculate_bonus(2800, true) == 1400
puts calculate_bonus(1000, false) == 0
puts calculate_bonus(50000, true) == 25000

algorithm:
- if bonus is true, return 1/2 the salary, if not, return 0


=end

File: exercises/101-109_small_problems/easy_01/rust/easy_10.rs
// What's my Bonus?
fn main() {
    println!("{}", calculate_bonus(2800, true) == 1400);
    println!("{}", calculate_bonus(1000, false) == 0);
    println!("{}", calculate_bonus(50000, true) == 25000);
}

fn calculate_bonus(salary: i32, bonus: bool) -> i32 {
    if bonus {
        salary / 2
    } else {
        0
    }
}
